### DISCUSSION TITLE
Validating Self-Signed Certificate

### DISCUSSION THREAD

**Discussion by brannonh** [sindresorhus/got#1729] [Q&A] [answered]
Hello,

I just implemented Got in a project where we are connecting to a partner's API. They are utilizing a self-signed certificate (behind a VPN tunnel) that I would like to validate, however they will only provide the client version of their certificate, citing security risks to providing the server version of the cert and their private key.

My question is am I able to validate the certificate with just the client version through Got? I would prefer not disabling certificate validation if I can help it.

Thanks!

**Comment by brannonh** [comment]
🦗 🦗 🦗 

I'm assuming by the silence that this is not currently supported. Is that accurate?

**Comment by Giotino** [comment, marked as answer]
You can use [https.checkServerIdentity](https://github.com/sindresorhus/got#httpscheckserveridentity).

**Reply by brannonh** [reply]
I saw that, but per the docs at that link...

> Note: In order to have the function called the certificate must not be `expired`, `self-signed` or with an `untrusted-root`.

My question is specifically about a `self-signed` certificate.

**Reply by szmarczak** [reply]
I haven't checked this. Have you tried it anyway?

**Reply by szmarczak** [reply]
I think you need to update trusted CAs https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options

**Reply by brannonh** [reply]
I did not check it, due to the docs. If there's uncertainty about whether or not the docs are accurate, I'm happy to test it. Will let you guys know.

**Reply by brannonh** [reply]
Okay, thanks, @szmarczak. Will look into that also.

**Reply by netapp-srinivas** [reply]
Hi @brannonh , we are trying to do something similar. Can you update what worked for you?
